lib/spack/spack/test/versions.py

```diff
@@ -730,3 +730,17 @@ def test_version_list_with_range_included_in_concrete_version_interpreted_as_ran
 def test_version_list_with_range_and_concrete_version_is_not_concrete():
     v = VersionList([Version('3.1'), VersionRange('3.1.1', '3.1.2')])
     assert v.concrete
+
+
+@pytest.mark.parametrize('vstring, eq_vstring',
+                         (('abc12' * 8 + '=develop', 'develop'),
+                          ('git.' + 'abc12' * 8 + '=main', 'main'),
+                          ('a' * 40 + '=develop', 'develop')
+                          ))
+def test_git_hash_can_be_assigned_a_version(vstring, eq_vstring):
+    v = Version(vstring)
+    v_equivalent = Version(eq_vstring)
+    assert v.is_commit
+    assert v.is_ref
+    assert not v._ref_lookup
+    assert v_equivalent.version[0] == v.ref_version
```

Comment:
This comparison only works if the equivalent version is a single component (`develop` or `main`, but not `3.2`). Can you modify this test to use a semver version for one of the comparison points?